CI Check Failure: Workflow expects calculator.tests.js but test file pattern differs

## Issue Description

The Step 3 CI workflow check failed because of a file naming mismatch in the test file path validation.

## Problem

The workflow file (`.github/workflows/3-step.yml`) is checking for the existence of `src/tests/calculator.tests.js`, but:

1. Jest test discovery typically looks for files matching the pattern `**/*.test.js` or `**/*.spec.js`
2. When a user follows common Node.js/Jest conventions and creates `calculator.test.js`, the workflow fails
3. The keyphrase checker throws: `##[error]File does not exist: src/tests/calculator.tests.js`
4. This causes all subsequent checks to fail even though the tests exist and pass

## Error Output

```
##[error]File does not exist: src/tests/calculator.tests.js
##[error]File does not exist: src/tests/calculator.tests.js
```

The test file exists at `src/tests/calculator.test.js` with all 61 tests passing.

## Root Cause

The workflow expects a non-standard test file naming convention (`calculator.tests.js` with double extension) instead of the standard (`calculator.test.js`), causing confusion and requiring users to rename files to match the workflow's expectations rather than following JavaScript/Jest conventions.

## Solution

Either:
1. Update the workflow to check for `src/tests/calculator.test.js` (standard Jest convention), or
2. Update the exercise instructions to explicitly require `calculator.tests.js` naming, or
3. Check for both file naming patterns in the workflow validation

## Steps to Reproduce

1. Create a calculator test file following Jest conventions: `src/tests/calculator.test.js`
2. Push changes to the repository
3. The Step 3 workflow CI check fails with file not found errors
4. Rename file to `calculator.tests.js` for workflow to pass

--- a/.github/steps/2-step.md
+++ b/.github/steps/2-step.md
@@ -128,7 +128,7 @@ Use the `!` command in Copilot CLI to execute shell commands directly from your 
    > Create comprehensive unit tests for all the calculator functions:
    > - Expand tests based on the following example:
    >   - @images/calc-basic-operations.png
-   > - Add these tests to a src/tests/calculator.tests.js file
+   > - Add these tests to a src/tests/calculator.test.js file
    > - Use a popular Node.js testing framework if one isn't installed
    > - addition, subtraction, multiplication, and division
    > - test edge cases like division by zero

--- a/.github/steps/3-step.md
+++ b/.github/steps/3-step.md
@@ -112,7 +112,7 @@ As you add features, Copilot CLI can help you:
    > Add tests for the new calculator operations: 
    > - Expand tests based on the following example:
    >   - @images/calc-extended-operations.png
-   > - Add new tests for the new operations to the existing src/tests/calculator.tests.js file
+   > - Add new tests for the new operations to the existing src/tests/calculator.test.js file
    > - Use a popular Node.js testing framework if one isn't installed
    > - Make sure to include edge case tests like square root of negative numbers
    > - Make sure all tests run and pass

--- a/.github/workflows/3-step.yml
+++ b/.github/workflows/3-step.yml
@@ -57,28 +57,40 @@ jobs:
           file: exercise-toolkit/markdown-templates/step-feedback/checking-work.md
           edit-mode: replace
 
-      - name: Check if calculator.tests.js has a modulo test
+      - name: Detect calculator test file
+        id: detect-test-file
+        run: |
+          if [ -f "src/tests/calculator.test.js" ]; then
+            echo "path=src/tests/calculator.test.js" >> "$GITHUB_OUTPUT"
+          elif [ -f "src/tests/calculator.tests.js" ]; then
+            echo "path=src/tests/calculator.tests.js" >> "$GITHUB_OUTPUT"
+          else
+            echo "::error::Expected src/tests/calculator.test.js or src/tests/calculator.tests.js but neither was found"
+            exit 1
+          fi
+
+      - name: Check if calculator test file has a modulo test
         id: check-modulo-test
         continue-on-error: true
         uses: skills/action-keyphrase-checker@v1
         with:
-          text-file: src/tests/calculator.tests.js
+          text-file: ${{ steps.detect-test-file.outputs.path }}
           keyphrase: modulo
 
-      - name: Check if calculator.tests.js has a power test
+      - name: Check if calculator test file has a power test
         id: check-power-test
         continue-on-error: true
         uses: skills/action-keyphrase-checker@v1
         with:
-          text-file: src/tests/calculator.tests.js
+          text-file: ${{ steps.detect-test-file.outputs.path }}
           keyphrase: power
 
-      - name: Check if calculator.tests.js has a 'square root' test
+      - name: Check if calculator test file has a 'square root' test
         id: check-sqrt-test
         continue-on-error: true
         uses: skills/action-keyphrase-checker@v1
         with:
-          text-file: src/tests/calculator.tests.js
+          text-file: ${{ steps.detect-test-file.outputs.path }}
           keyphrase: 'square'
 
       - name: Update comment - step results

